service/src/main/java/bio/terra/cbas/controllers/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.cbas.controllers;
+
+import bio.terra.cbas.model.ErrorReport;
+import bio.terra.common.exception.AbstractGlobalExceptionHandler;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+/**
+ * <a
+ * href="https://github.com/DataBiosphere/terra-java-project-template/blob/43cff86354e8d295ed96c29a7faf60eadd11e502/service/src/main/java/bio/terra/javatemplate/controller/GlobalExceptionHandler.java">Taken
+ * from Terra template</a> This enables proper logging and response codes when errors are thrown
+ * within the code (primarily when interacting with Sam). Without this, any errors are reported to
+ * the user as a 500 Internal Server Error.
+ */
+@RestControllerAdvice
+public class GlobalExceptionHandler extends AbstractGlobalExceptionHandler<ErrorReport> {
+
+  @Override
+  public ErrorReport generateErrorReport(Throwable ex, HttpStatus statusCode, List<String> causes) {
```

Comment:
I am having a little trouble understanding where this is used. Can you point me to it? Does this apply to all APIs?

Reply:
My understanding is that this is automatically hooked into controllers and called when controller methods throw exceptions, which is useful because not all of our error cases return nice ResponseEntities, but we don't want every exception classified as an InternalServerError.

Reply:
It is similar to the BearerTokenFilter in that we do not explicitly call it, but Spring will call this method on outgoing responses with Exceptions. Here's a link with more info: [Spring Exception handling](https://www.baeldung.com/exception-handling-for-rest-with-spring), check section titled "Solution 3". This is used in many of the other Terra java projects as well for reference